### ISSUE TITLE
regression: 2.12.3 source compat issue with `Namers._lockedCount`

### ISSUE THREAD

**Issue by SethTisue** [scalamacros/paradise#98]
in scala/scala@47d35913c1aa16c1f3fb3bc4c3166daa0df93af8 @retronym renamed `_lockedCount` which paradise is using ~~in the `test` subproject~~:

```
[macro-paradise] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/macro-paradise-def0457506293f04c71773e96e1046e7020b438b/plugin/src/main/scala/org/scalamacros/paradise/typechecker/Namers.scala:322: not found: value _lockedCount
[macro-paradise] [error]         _lockedCount += 1
[macro-paradise] [error]         ^
[macro-paradise] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/macro-paradise-def0457506293f04c71773e96e1046e7020b438b/plugin/src/main/scala/org/scalamacros/paradise/typechecker/Namers.scala:324: not found: value _lockedCount
[macro-paradise] [error]         finally _lockedCount -= 1
[macro-paradise] [error]                 ^
[macro-paradise] [error] two errors found
```

as seen at https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1480/consoleFull

**Comment by SethTisue** [comment]
(I mistakenly thought this was just in a subproject I could disable, but no, it's in the main code)

**Comment by SethTisue** [comment]
reproducible outside the community build with the https://scala-ci.typesafe.com/artifactory/scala-integration/ resolver and:

```
> ++2.12.3-bin-20751a2
> paradise/compile
...
[error] /Users/tisue/paradise/plugin/src/main/scala/org/scalamacros/paradise/typechecker/Namers.scala:322: not found: value _lockedCount
[error]         _lockedCount += 1
[error]         ^
[error] /Users/tisue/paradise/plugin/src/main/scala/org/scalamacros/paradise/typechecker/Namers.scala:324: not found: value _lockedCount
[error]         finally _lockedCount -= 1
[error]                 ^
```

**Comment by xeno-by** [comment]
Hmm.

When's the release date for 2.12.3? I'm thinking that instead of fixing the divergencies over and over again, I should just try and submit a PR that folds paradise into scalac as discussed with @adriaanm.

**Comment by adriaanm** [comment]
The release is schedule for end of June. For a bigger change like this, we'd have to get the PR in the first week of June.

**Comment by SethTisue** [comment]
working around it for now in the community build with https://github.com/scalacommunitybuild/paradise/commit/f025f9cacea47916b2c64ad7213adc6b1b025378 (@xeno-by which you might want to grab onto a 2.12.3 branch) and https://github.com/scala/community-builds/commit/78c12c8782a94e4122481e1da5bb192354302e77

**Comment by milessabin** [comment]
I'd be happy to merge a PR folding paradise into scalac into Typelevel Scala and make a release more or less immediately.

**Comment by SethTisue** [comment]
fixed (a long while back) in 2c7f071ecd1d0e75b678fbdc81fc0e6e80042edf
